adagrams/game.py

```diff
@@ -1,11 +1,104 @@
+import random
+
 def draw_letters():
-    pass
+    letter_pool = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    final_letter_choice = []
+    for number in range(0,10):
+        chosen_letter = random.choice(list(letter_pool.keys()))
+        if (letter_pool[chosen_letter] == 1):
+            letter_pool.pop(chosen_letter)
+        else:
+            letter_pool[chosen_letter] -= 1
+        final_letter_choice.append(chosen_letter)
+        
+    return final_letter_choice
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_new = letter_bank.copy()
+    for characters in word:
+        try:
+            letter_bank_new.remove(characters.upper())
+        except ValueError:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    letter_score = {
+        'A': 1, 
+        'B': 3, 
+        'C': 3, 
+        'D': 2, 
+        'E': 1, 
+        'F': 4, 
+        'G': 2, 
+        'H': 4, 
+        'I': 1, 
+        'J': 8, 
+        'K': 5, 
+        'L': 1, 
+        'M': 3, 
+        'N': 1, 
+        'O': 1, 
+        'P': 3, 
+        'Q': 10, 
+        'R': 1, 
+        'S': 1, 
+        'T': 1, 
+        'U': 1, 
+        'V': 4, 
+        'W': 4, 
+        'X': 8, 
+        'Y': 4, 
+        'Z': 10
+    }
+    final_score = 0
+    for characters in word:
+        final_score += letter_score[characters.upper()]
+    if len(word) in range(7, 11):
```

Comment:
[nit] `if len(word) >= 7:` might be less unexpected

Reply:
I noticed you changed away from this suggestion in one of your commits. The commit message only says "Another change", so I can't tell what the reasoning was behind the change. A more descriptive commit message might have prevented me from leaving this code review comment, which is excellent motivation for having more descriptive commit messages. 😉